src/util.js

```diff
@@ -5,5 +5,5 @@ export function parseDateTime(datetime) {
   let [hour, minute, second] = time.split(":");
   let [day, month, year] = date.split("-");
 
-  return new Date(year, parseInt(month)-1, day, hour, minute, second);
+  return new Date(Date.UTC(year, parseInt(month)-1, day, hour, minute, second));
```

Comment:
The string format you are seeing is the default format that JavaScript uses when converting it's format to String. They are already Date objects. Currently Home Assistant assumes client and server are in the same time zone. All dates/times are stored without timezone. This should be refactored, will probably take quite some effort.